### PR TITLE
Add termcolor-cpp feedstock

### PR DESCRIPTION
**Pull request by adriendelsalle**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/termcolor-cpp/bld.bat [hunk 1]
```diff
@@ -0,0 +1,8 @@
+mkdir build
+cd build
+
+cmake -G "NMake Makefiles" -D CMAKE_BUILD_TYPE=Release -D CMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ..
```
Comment (wolfv) [inline recipes/termcolor-cpp/bld.bat:4]: might be useful to use `%CMAKE_ARGS%` as well, just in case

File: recipes/termcolor-cpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{% set version = "2.0.0" %}
+
+package:
+  name: termcolor-cpp
+  version: {{ version }}
+
+source:
+  fn: termcolor-{{ version }}.tar.gz
```
Comment (wolfv) [inline recipes/termcolor-cpp/meta.yaml:8]: not needed.

Comment (adriendelsalle) [reply recipes/termcolor-cpp/meta.yaml:8]: good catch, thx, typo from copy pasting

File: recipes/termcolor-cpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+test:
+  commands:
+    - test -e $PREFIX/include/termcolor/termcolor.hpp  # [not win]
+    - if not exist %PREFIX%\Library\include\termcolor\termcolor.hpp exit 1  # [win]
+
+about:
+  home: https://github.com/ikalnytskyi/termcolor
+  license: BSD-3-Clause
+  license_family: BSD
```
Comment (wolfv) [inline recipes/termcolor-cpp/meta.yaml:29]: not needed as well

**Comment by wolfv** [comment]
Looks great!

File: recipes/termcolor-cpp/meta.yaml (hunk 3 above)
Comment (adriendelsalle) [reply recipes/termcolor-cpp/meta.yaml:29]: I just keep license?

File: recipes/termcolor-cpp/bld.bat (hunk 1 above)
Comment (adriendelsalle) [reply recipes/termcolor-cpp/bld.bat:4]: only for windows build script? not needed on unix?

Comment (wolfv) [reply recipes/termcolor-cpp/bld.bat]: you already have it on UNIX :)

**Pull request merged**